warhammer-core/WarhammerCore.Business/UserService.cs

```diff
@@ -9,17 +15,27 @@ namespace WarhammerCore.Business
     public class UserService : IUserService
     {
         private readonly IDataRepo _repo;
-        public UserService(IDataRepo repo)
+        private readonly IConfiguration _config;
+        public UserService(IDataRepo repo,
+            IConfiguration config
+            )
         {
             _repo = repo;
+            _config=config;
         }
 
         /// <summary>
         /// <inheritdoc/>
         /// </summary>
         public async Task<UserAuthInfo> SignInAsync(string email, string password)
         {
-            throw new NotImplementedException();
+          var user =   await _repo.GetUserByEmailAsync(email);
+
+            if (user.UserId == UserInfo.Empty.UserId) return null;
+
+            var auth = new UserAuthInfo(user.UserId,GenerateJSONWebToken(user));
+
+            return auth;
```

Comment:
Please install [Code Maid Extension](https://marketplace.visualstudio.com/items?itemName=SteveCadwallader.CodeMaid) and format the code so it's nicely formatted.

![image](https://user-images.githubusercontent.com/43864376/104847996-4b5ae280-58e3-11eb-8d65-75d2fff57cd0.png)


```suggestion
            var user = await _repo.GetUserByEmailAsync(email);

            if (user.UserId == UserInfo.Empty.UserId) return null;

            var auth = new UserAuthInfo(user.UserId, GenerateJSONWebToken(user));

            return auth;
```